WD-17111 - Implement new data/streaming page and form

pedoch opened the pull request:
## Done

Implement new page: `/data/streaming`

## QA

- Navigate to /data/streaming
- Ensure the design is aligned with the Figma file. 
- Ensure the page content aligns with the copy doc

## Issue / Card

- [WD-17111](https://warthogs.atlassian.net/browse/WD-17111)

webteam-app commented:
### [<img src='https://assets.ubuntu.com/v1/6baef514-ubuntu-circle-of-friends-large.svg' height=32 width=32> Demo</img>](https://canonical-com-1470.demos.haus)
### [<img src='https://assets.ubuntu.com/v1/e512b0e2-jenkins.svg' height=32 width=32> Jenkins </img>](https://jenkins.canonical.com/webteam/job/start-demo/7673)
### [<img src='https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg' height=32 width=32> demos.haus </img>](https://demos.haus)

mattea-turic commented:
Thanks @pedoch ! A couple of comments:

For the hero section:
- [ ] I noticed that the image is a little too light at the top (blends with bg), so I've added an updated asset [here](https://assets.ubuntu.com/v1/bca3146e-hero-img-updated.png)
<br>

For the "Streaming use cases" section:
- [ ] Could you wrap the 50/50 subsection (text + video) in a shallow pls?
- [ ] And use a muted hr above the grid of images
<br>

For the "Canonical solutions for data streaming" section:
- [ ] On mobile, there's spacing lacking between the two pricing cards
<br> 

For the "resources" section:
- [ ] The hrs that don't span full-width should be muted
<br>

For the footer:
- [ ] I think you've mistakenly duplicated the trademark text

pedoch commented:
Thanks for the review @mattea-turic. I've made the changes that were requested. 

A couple of things to note:
- While I added space on the pricing cards, I checked for other examples where this was used, and the cards on other pages don't have a space between them on mobile screens. So, this is kind of a "custom solution" for this page.
- The trademark texts are actually different; they are very similar but refer to different products. This is how it's stated in the copy. They could be merged, but that must be changed on the copy doc.

mattea-turic commented:
Hey @pedoch , the link for the page doesn't seem to be working (https://canonical-com-1470.demos.haus/data/streaming) for me

pedoch commented:
@mattea-turic looks like demos aren't coming up. I'll let you know as soon as it's up.

pedoch commented:
@mattea-turic the demo is up now

mattea-turic commented:
Thanks @pedoch , LGTM! I'll +1 for design

eliman11 commented:
Thanks @pedoch looks great! Nothing missing from UX once we have the Marketo ID and whitepaper link you've pinged about already.

muhammad-ali-pk commented on templates/data/streaming.html:
```diff
@@ -0,0 +1,429 @@
+
+{% block content %}
+
+  <section class="p-section--hero">
+    <div class="row--50-50">
+      <div class="col">
+        <h1>What is Data streaming?</h1>
+      </div>
+      <div class="col p-section--shallow">
+        <div class="p-section--shallow">
+					<p>
+						Data streaming is the continuous processing of data as it is generated. It is also known as event processing or stream processing. The benefit of data streaming is that analytics, insights and actions can occur immediately. Data streaming is used across industries like financial services, retail, media and entertainment, and telecommunications.
+					</p>
+        </div>
+        <hr class="p-rule--muted" />
+        <p>
+          <a href="#" class="p-button--positive">Download the white paper</a>
+        </p>
+      </div>
+      <div class="p-image-container--cinematic is-highlighted u-hide--small">
+				{{ image(url="https://assets.ubuntu.com/v1/bca3146e-hero-img-updated.png",
+										alt="",
+										width="2464",
+										height="1027",
+										hi_def=True,
+										loading="lazy",
+										attrs={"class": "p-image-container__image"}) | safe
+				}}
+      </div>
+    </div>
+  </section>
```
```suggestion
    {% call(slot) vf_hero(
        title_text='What is Data streaming?',
        layout='50/50-full-width-image'
        ) -%}
        {%- if slot == 'description' -%}
            <p>
                Data streaming is the continuous processing of data as it is generated. It is also known as event processing or stream processing. The benefit of data streaming is that analytics, insights and actions can occur immediately. Data streaming is used across industries like financial services, retail, media and entertainment, and telecommunications.
            </p>
        {%- endif -%}
        {%- if slot == 'cta' -%}
            <a href="#" class="p-button--positive">Download the white paper</a>
        {%- endif -%}
        {%- if slot == 'image' -%}
            <div class="p-image-container--cinematic is-cover">
                {{ image(url="https://assets.ubuntu.com/v1/bca3146e-hero-img-updated.png",
                                alt="",
                                width="2464",
                                height="1027",
                                hi_def=True,
                                loading="lazy",
                                attrs={"class": "p-image-container__image"}) | safe
                }}
            </div>
        {% endif -%}
    {% endcall -%}
```

We try to use Vanilla patterns where we could. This hero section is one such example.
Please make sure to import the following script on the top, after `extends`
`{% from "_macros/vf_hero.jinja" import vf_hero %}`

muhammad-ali-pk replied:
Please also update Vanilla version to 4.18.5 or latest

muhammad-ali-pk commented:
Please also update this branch w.r.t. to main to resolve conflicts.

muhammad-ali-pk commented:
Please also reformat the entire file with djlint.

akbarkz commented:
@pedoch Can you please address Muhammad's comments?

pedoch commented:
Hi @akbarkz I have started working on addressing comments and will push the changes later today. I haven't prioritised this task as we're still missing information from the PM which is blocking this from getting merged. I left [a comment on the task](https://warthogs.atlassian.net/browse/WD-17111?focusedCommentId=653333) about this. No response yet.

pedoch commented:
Hi @muhammad-ali-pk, this is ready for your review again. Incase you want to take another look.

muhammad-ali-pk reviewed (approved):
LGTM!

muhammad-ali-pk commented:
@pedoch Looks like the file was not properly formatted with djlint on your end for some reason. I have formatted it for you and cleaned up 2 links. The rest is good! Thanks!

pedoch commented:
@muhammad-ali-pk the job failed because the branch wasn't up to date with main. I've updated it.

eliman11 commented:
Hey @pedoch thank you for your patience on this and apologies for the back and forth. I just spoke to Giulia and agreed we should have "Get in touch" with a link to the same contact form instead of the white paper as the one being linked atm isn't as relevant to the content. I've added it as a suggestion with my comment on the copydoc - appreciate your help!

pedoch commented:
@eliman11 the CTA on the hero now opens the form. I’ve tested the form and can see that the submission works. This should be good to go.
